Contest_API.md

```diff
@@ -916,11 +916,11 @@ Properties of language objects:
 | :------------------- | :--------------- | :----------
 | id                   | ID               | Identifier of the language from table below.
 | name                 | string           | Name of the language (might not match table below, e.g. if localized).
-| entry_point_required | boolean ?        | Whether the language requires an entry point. Defaults to `false`.
+| entry_point_required | boolean          | Whether the language requires an entry point. Defaults to `false`.
```

Comment:
If this property is declared in `/access`, then I think it's reasonable to require that *each* language has it explicitly set (can still be either `false` or `true`).

Reply:
I'm torn about this one. Missing (i.e. null) instead of false is a nice shorthand?

Reply:
I'd really prefer to be explicit instead of allowing shorthand alternatives. If you already declared this property in `/access`, then you most likely have some languages for which `entry_point_required=true`, so I don't think it saves much to be able to leave the attribute out, and that it doesn't weigh up to the clarity of explicitly having it present.

Reply:
I agree with Jaap. I also can't think of an implementation (either server or client) that would be easier if we do this. I think it only saves bytes on the wire, which should not be a reason IMHO.

Reply:
Ok, yeah, those are good points. I agree.

Reply:
> One thing I noticed earlier, which is somewhat related:
> 
> For `frozen` and `thawed` in the state endpoint we say something like:
> 
> > Required iff scoreboard_freeze_duration is present in the [contest](https://ccs-specs.icpc.io/master/contest_api#contests) endpoint.
> 
> But the freeze duration can be present but `0:00:00` which is the same as it not being present. We should maybe reword this?

Agreed. I think we should reword that part so that the value `0:00:00` is explicitly forbidden and `null` means "no freeze".